lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -502,8 +502,7 @@ void TypeLoweringVisitor::lowerSAWritePath(Operation *op,
                                            ArrayRef<Operation *> writePath) {
   SubaccessOp sao = cast<SubaccessOp>(writePath.back());
   auto saoType = sao.input().getType().cast<FVectorType>();
-  auto selectWidth =
-      sao.index().getType().cast<FIRRTLType>().getBitWidthOrSentinel();
+  auto selectWidth = llvm::Log2_64_Ceil(saoType.getNumElements());
```

Comment:
Should this be max to avoid implicit truncation when the index is too big?

Reply:
These lead to `firrtl.eq` with different bit width operands (one has the width of the index, one the clog2 of the vector length), which implicitly sign/zero-extends the operands to `max(lhs, rhs)` before doing the equality. So this should do what we want :+1: